### ISSUE TITLE
How do i use proxy while i create a client?

### ISSUE THREAD

**Issue by dlsagayaraj** [gildean/node-flowdock-stream#2]
Below is my code for creating a client to listen to the flowdock stream. But how do i create a client with proxy ?

var flowdockStream = FlowdockStream.createClient(org, flows, apikey);

**Comment by gildean** [comment]
Hi,
I just published an update (v.1.0.3), which should fix the https-proxy support.
The proxy should be set to `HTTPS_PROXY` (or perhaps `HTTP_PROXY` depending on the env) environment variable. See here: https://github.com/request/request#controlling-proxy-behaviour-using-environment-variables

**Comment by gildean** [comment]
Closing this as presuming fixed.

**Comment by dlsagayaraj** [comment]
Hi, 
  i tried with v.1.0.3, but didnt help. I tried adding HTTPS_PROXY/ HTTP_PROXY to the process.env of the node but it didnt work for me. I also tried by setting the HTTPS_PROXY in my linux terminal which didnt work either. I could connect through proxy through for other API like zendesk. Any thoughts on this? 

This is how i create my zendesk client. 

```
 zendeskClient = zendesk.createClient({
        username: config.zendesk.username,
        token: config.zendesk.token,
        remoteUri: config.zendesk.url + config.zendesk.apiVersion
        ,proxy: config.zendesk.proxy
    });
```

Is there a way that i could mention the proxy on the client creation instead of relying on the env variables.

**Comment by gildean** [comment]
Hi,

I just released version 1.1.1 which allows you to pass an object with request-defaults as the fourth argument to the constructor.
Update and see the readme for help or for more info on the options, see: https://github.com/request/request#proxies
As far as I could test this, it should now work reliably, but I don't have a real enterprise-class proxy at hand to test this.

**Comment by gildean** [comment]
Presuming fixed.
